### PR TITLE
remove ReliableOperation

### PR DESCRIPTION
**Pull request by JoshLove-msft**
The same functionality is available out of the box in Core Pipeline (resetting the stream to 0).
There are other slight differences in how ReliableOperation was implemented:
- it retried on all exceptions except for OperationCanceled - but we don't want this - we should use ResponseClassifier
- it retried up to a max of 15 times - we don't want this - it should use RetryOptions.

Also remove remnant of queue cryptography project from soln file that was removed in https://github.com/Azure/azure-sdk-for-net/commit/9250bd0c1a3f1b20a76bcf5dcd47709635d04dbe#diff-3d4d8346908ac33138aadb34bb3bf2a3

### CODE REVIEW

**Review by seanmcc-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs [hunk 1]
```diff
@@ -486,40 +485,34 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
...
-                                ifMatch: conditions?.IfMatch,
-                                ifNoneMatch: conditions?.IfNoneMatch,
-                                async: async,
-                                operationName: Constants.Blob.Block.UploadOperationName,
-                                cancellationToken: cancellationToken)
-                                .ConfigureAwait(false);
-                        },
-                        new ReliabilityConfiguration(reset: () => content.Seek(0, SeekOrigin.Begin)))
-                        .ConfigureAwait(false);
+                    BlobErrors.VerifyHttpsCustomerProvidedKey(Uri, CustomerProvidedKey);
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs]: ???

File: sdk/storage/Azure.Storage.Blobs/src/PageBlobClient.cs [hunk 2]
```diff
@@ -814,140 +814,128 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
                 {
                     content = content.WithNoDispose().WithProgress(progressHandler);
                     var range = new HttpRange(offset, content.Length);
-                    var uploadAttempt = 0;
-                    return await ReliableOperation.DoSyncOrAsync(
-                        async,
-                        reset: () => content.Seek(0, SeekOrigin.Begin),
-                        predicate: e => true,
-                        maximumRetries: Constants.MaxReliabilityRetries,
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Blobs/src/PageBlobClient.cs]: Is the `MaxReliabilityRetries` still used anywhere?

File: sdk/storage/Azure.Storage.Blobs/src/PageBlobClient.cs [hunk 3]
```diff
@@ -1033,998 +1021,488 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
...
+    /// </summary>
+    /// <param name="range">
+    /// Optionally specifies the range of bytes over which to list ranges,
+    /// inclusively. If omitted, then all ranges for the blob are returned.
+    /// </param>
+    /// <param name="snapshot">
+    /// Optionally specifies the blob snapshot to retrieve page ranges
+    /// information from. For more information on working with blob snapshots,
+    /// <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/creating-a-snapshot-of-a-blob"/>.
+    /// </param>
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Blobs/src/PageBlobClient.cs]: Unto extra changes.

File: sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs]: Merge error

File: sdk/storage/Azure.Storage.Blobs/src/PageBlobClient.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Blobs/src/PageBlobClient.cs]: Yeah it is used in our tests, but probably shouldn't be.

**Review by pakrym — APPROVED** [review]
Pending merge fix.

**Pull request merged**